hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -1201,28 +1322,38 @@ void XmlChangeset::replaceRelationId(long old_id, long new_id)
   _idMap.updateId(ElementType::Relation, old_id, new_id);
 }
 
-void XmlChangeset::failNode(long id)
+void XmlChangeset::failNode(long id, bool beforeSend)
 {
   //  Set the node's status to failed
   _allNodes[id]->setStatus(XmlElement::ElementStatus::Failed);
   //  Update the failed count once
   ++_failedCount;
+  //  Update sent count as if we already sent it and it failed
+  if (beforeSend)
+    ++_sentCount;
```

Comment:
When pre-upload analysis finds a node/way/relation that it knows will fail without even having to send it, the status is set to failed, the failed count is increased, and the sent count is increased.  This allows the changeset splitting algorithm to know if all elements have been sent to the OSM API.